pandas/core/indexes/api.py

```diff
@@ -311,3 +311,24 @@ def all_indexes_same(indexes):
         if not first.equals(index):
             return False
     return True
+
+
+def transform_index(index: Index, func, level=None) -> Index:
```

Comment:
how is this different from .map?

Reply:
Looks like the main difference is in the MultiIndex case passing level makes this operate on just that levle

Reply:
k, it would be better to use map then than duplicating this code

Reply:
or at least co-locate as a private function for future de-duplication

Reply:
Updated